ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/PurchasesType.kt

```diff
@@ -0,0 +1,44 @@
+package com.revenuecat.purchases.ui.revenuecatui.data
+
+import com.revenuecat.purchases.CacheFetchPolicy
+import com.revenuecat.purchases.CustomerInfo
+import com.revenuecat.purchases.Offerings
+import com.revenuecat.purchases.PurchaseParams
+import com.revenuecat.purchases.PurchaseResult
+import com.revenuecat.purchases.Purchases
+import com.revenuecat.purchases.awaitOfferings
+import com.revenuecat.purchases.awaitPurchase
+import com.revenuecat.purchases.awaitRestore
+
+/**
+ * Abstraction over [Purchases] that can be mocked.
+ */
+internal interface PurchasesType {
+    suspend fun awaitPurchase(purchaseParams: PurchaseParams.Builder): PurchaseResult
+
+    suspend fun awaitRestore(): CustomerInfo
+
+    suspend fun awaitOfferings(): Offerings
+
+    suspend fun awaitCustomerInfo(
+        fetchPolicy: CacheFetchPolicy = CacheFetchPolicy.default(),
+    ): CustomerInfo
+}
+
+internal class PurchasesImpl(private val purchases: Purchases = Purchases.sharedInstance) : PurchasesType {
+    override suspend fun awaitPurchase(purchaseParams: PurchaseParams.Builder): PurchaseResult {
+        return purchases.awaitPurchase(purchaseParams.build())
+    }
+
+    override suspend fun awaitRestore(): CustomerInfo {
+        return purchases.awaitRestore()
+    }
+
+    override suspend fun awaitOfferings(): Offerings {
+        return purchases.awaitOfferings()
+    }
+
+    override suspend fun awaitCustomerInfo(fetchPolicy: CacheFetchPolicy): CustomerInfo {
+        return awaitCustomerInfo(fetchPolicy)
```

Comment:
I think this infinite loop still exists?

Reply:
OMG yes haha, let me check why it wasn't caught. I never dealt with this because I deleted this type until we realized we did need it.